sdks/java/io/amazon-web-services/src/main/java/org/apache/beam/sdk/io/aws/sns/SnsCoderProviderRegistrar.java

```diff
@@ -32,6 +32,7 @@
   @Override
   public List<CoderProvider> getCoderProviders() {
     return ImmutableList.of(
-        CoderProviders.forCoder(TypeDescriptor.of(PublishResult.class), PublishResultCoder.of()));
+        CoderProviders.forCoder(
+            TypeDescriptor.of(PublishResult.class), PublishResultCoders.defaultPublishResult()));
```

Comment:
Same here, any reasons this is not `XXXcoder.of(...)` pattern?

Reply:
It's sort of the same reason as above, PublishResultCoders is a factory for creating several different coder implementations so the factory methods have different names to indicate which one. I could rename this method to `of()` but since we are introducing new behavior I wanted to make it very clear that this is the default, or existing, behavior. Also I'm not sure if it makes sense to have a method called `of()` on the factory not on the coder itself.